### PR TITLE
feat(content): add origin vary and cors headers to .lnk requests

### PR DESCRIPTION
**Pull request by trieloff**
fixes (maybe) #830

### CODE REVIEW

File: layouts/fastly/helix.vcl [hunk 1]
```diff
@@ -1101,6 +1101,19 @@ sub hlx_fetch_query {
...
 }
 
+sub hlx_fetch_lnk {
+  if (req.http.X-Request-Type == "Content/LNK") {
+    if (!req.is_esi_subreq) {
+      set beresp.http.X-PostFetch = beresp.http.X-PostFetch + "; hlx_fetch_lnk";
+    }
+    if (req.http.X-Orig-URL ~ "hlx_report=true") {
+      set beresp.http.Access-Control-Allow-Origin = req.http.Origin;
+      set beresp.http.Access-Control-Allow-Methods = "GET, OPTIONS, POST";
```
Comment (rofe) [inline layouts/fastly/helix.vcl:1111]:
```suggestion
      set beresp.http.Access-Control-Allow-Methods = "GET, OPTIONS";
```
`POST` is not needed for .lnk requests

**Review by rofe — APPROVED** [review]

File: layouts/fastly/helix.vcl [hunk 2]
```diff
@@ -2359,11 +2374,14 @@ sub vcl_deliver {
     set req.http.x-last-activation-id = resp.http.x-last-activation-id;
   }
 
-  if (!req.http.X-Debug) {
-    # Unless we are debugging, shut up chatty headers
+  if (!req.http.X-Debug && req.http.X-Request-Type != "Content/LNK") {
```
Comment (tripodsan) [inline layouts/fastly/helix.vcl:2377]: why would be ever unset those? 
right now, the content proxy correctly sets the cors headers. so no need to recompute them here.

File: layouts/fastly/helix.vcl (hunk 1 above)
Comment (tripodsan) [reply layouts/fastly/helix.vcl:1111]: I would remove all this and just allow the CORS headers from the backend to be honoured.

**Review by tripodsan — CHANGES_REQUESTED** [review]

File: layouts/fastly/helix.vcl (hunk 2 above)
Comment (trieloff) [reply layouts/fastly/helix.vcl:2377]: We are serving from a number of backends that are all opinionated about CORS, including Adobe I/O and GitHub. The generic cleanup is the best approach to deal with that.

**Comment by trieloff** [comment]
Obsolete, check #871 instead.

**Pull request closed without merging**